e2e/cypress/integration/bot_accounts/crud_spec.js

```diff
@@ -0,0 +1,252 @@
+            cy.wrap(el[0].parentElement.parentElement).findByText('Edit').should('be.visible').click();
+
+            // * Validate redirect to edit screen
+            cy.url().
+                should('include', `/${newTeam.name}/integrations/bots/edit`);
+
+            // # type long string
+            const longDescription = 'A'.repeat(1020); // 1024 is the limit
+            cy.get('#description').clear().type(longDescription);
+
+            // * Validate that it's fully typed
+            cy.get('#description').should('have.value', longDescription);
+
+            // # type some more characters
+            cy.get('#description').type('{end}12345');
+
+            // * Validate that it's partially updated
+            cy.get('#description').should('have.value', longDescription + '1234');
+
+            // # Update the bot
+            cy.get('#saveBot').click();
+
+            // * Validate that bot saved correctly
+            cy.url().should('include', `/${newTeam.name}/integrations/bots`);
+        });
+    });
+
+    it('MM-T1842 Change BOT role', () => {
+        // # Visit the integrations
+        cy.visit(`/${newTeam.name}/integrations/bots`);
+
+        // * Check that the previously created bot is listed
+        cy.findByText(`Test Bot (@${botName})`).then((el) => {
+            // # Make sure it's on the screen
+            cy.wrap(el[0].parentElement.parentElement).scrollIntoView();
+
+            // # Click the edit button
+            cy.wrap(el[0].parentElement.parentElement).findByText('Edit').should('be.visible').click();
+
+            // * Validate redirect to edit screen
+            cy.url().
+                should('include', `/${newTeam.name}/integrations/bots/edit`);
+
+            // # Select sysadmin
+            cy.get(':nth-child(5) > .col-md-5 > .form-control').select('System Admin');
+
+            // * Validate that permissions are set and read only
+            cy.get('#postChannels').should('be.checked').should('be.disabled');
+            cy.get('#postAll').should('be.checked').should('be.disabled');
+
+            // # Update the bot
+            cy.get('#saveBot').click();
+
+            // * Validate that bot saved correctly
+            cy.url().should('include', `/${newTeam.name}/integrations/bots`);
+        });
+    });
+
+    function createBotInteractive(username = `bot-${getRandomId()}`) {
+        // # Visit the integrations
+        cy.visit(`/${newTeam.name}/integrations/bots`);
+
+        // # Click add bot
+        cy.get('#addBotAccount').click();
+
+        // # fill+submit form
+        cy.get('#username').type(username);
+        cy.get('#displayName').type('Test Bot');
+        cy.get('#saveBot').click();
+
+        // * verify confirmation page
+        cy.url().
+            should('include', `/${newTeam.name}/integrations/confirm`).
+            should('match', /token=[a-zA-Z0-9]{26}/);
+
+        // * verify confirmation form/token
+        cy.get('div.backstage-form').
```

Comment:
Is there any better way to access this element?

Reply:
not that I am aware of :(